skt/executable.py

```diff
@@ -97,6 +97,33 @@ def remove_oldresult(output_dir, prefix_filename):
         pass
 
 
+def update_state(state_file, state_dict):
```

Comment:
I have actually not checked the changes, but do you want to leverage `skt/state_file.py` for this?

Reply:
@veruu We could. I can move that if it makes more sense.

Reply:
Well, we already have that code and it isn't used anywhere, so changing it to fit the functionality we want instead of adding new stuff would be better, IMHO.

Reply:
@veruu  Done! I removed the unused functions/tests, too.